mypyc/irbuild/expression.py

```diff
@@ -33,12 +33,18 @@
 from mypyc.irbuild.for_helpers import translate_list_comprehension, comprehension_helper
 
 
-# Name and attribute references
+builtin_names = {
+    'builtins.dict': (object_rprimitive, 'PyDict_Type')
+}  # type: Dict[str, Tuple[RType, str]]
```

Comment:
Did you consider storing these inside a registry under `mypyc.primitives`? By declaring these there, we'd have the benefit of having most things related to a built-in type defined in a single place. The declaration of a name reference could be simplified like this, for example:

```py
load_address_op('builtins.dict', object_rprimitive, 'PyDict_Type')
```